lib/srv/app/cloud.go

```diff
@@ -185,31 +186,38 @@ func (c *cloud) getAWSSigninToken(ctx context.Context, req *AWSSigninRequest, en
 		return "", trace.Wrap(err)
 	}
 
-	options = append(options, func(creds *stscreds.AssumeRoleProvider) {
+	assumeRole := awsconfig.AssumeRole{
+		RoleARN:    req.Identity.RouteToApp.AWSRoleARN,
+		ExternalID: req.ExternalID,
 		// Setting role session name to Teleport username will allow to
 		// associate CloudTrail events with the Teleport user.
-		creds.RoleSessionName = awsutils.MaybeHashRoleSessionName(req.Identity.Username)
-
-		// Setting web console session duration through AssumeRole call for AWS
-		// sessions with temporary credentials.
-		// Technically the session duration can be set this way for
-		// non-temporary sessions. However, the AssumeRole call will fail if we
-		// are requesting duration longer than the maximum session duration of
-		// the role we are assuming. In addition, the session credentials may
-		// not have permission to perform a get-role on the role. Therefore,
-		// "SessionDuration" parameter will be defined when calling federation
-		// endpoint below instead of here, for non-temporary sessions.
-		//
-		// https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html
-		if temporarySession {
-			creds.Duration = duration
-		}
-
-		if req.ExternalID != "" {
-			creds.ExternalID = aws.String(req.ExternalID)
-		}
-	})
-	stsCredentials, err := stsutils.NewCredentialsV1(session, req.Identity.RouteToApp.AWSRoleARN, options...).Get()
+		SessionName: awsutils.MaybeHashRoleSessionName(req.Identity.Username),
+	}
+
+	// Setting web console session duration through AssumeRole call for AWS
+	// sessions with temporary credentials.
+	// Technically the session duration can be set this way for
+	// non-temporary sessions. However, the AssumeRole call will fail if we
+	// are requesting duration longer than the maximum session duration of
+	// the role we are assuming. In addition, the session credentials may
+	// not have permission to perform a get-role on the role. Therefore,
+	// "SessionDuration" parameter will be defined when calling federation
+	// endpoint below instead of here, for non-temporary sessions.
+	//
+	// https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html
+	if temporarySession {
+		assumeRole.Duration = duration
+	}
+
+	awsCfg, err := c.cfg.AWSConfigProvider.GetConfig(ctx, region,
+		awsconfig.WithDetailedAssumeRole(assumeRole),
+		awsconfig.WithCredentialsMaybeIntegration(req.Integration),
+	)
```

Comment:
aren't we using the cache provider by default? i think duration will cause issues with the cache. IMO, we shouldn't cache in this scenario. what do you guys think?

Reply:
I am also ok with not caching when `Duration` is set, just feels it's complicating the cache logic. We should still use cache for the first base config though but using `awsconfig.GetConfig` directly bypasses the cache completely? personally i would just call `stscreds.NewAssumeRoleProvider` here. but no strong opinion